### ISSUE TITLE
Remove need to use generated reflector getters/setters in element injector

### ISSUE THREAD

**Issue by yjbanov** [angular/angular#3436]
This will allow us to stop generating getters/setters, which are polymorphic, untyped, slow to execute (they're closures) and contribute to code size (cannot be minified as they use string names).

**Comment by kegluneq** [comment]
fyi https://github.com/kegluneq/angular/commits/remove_getters_setters is a start at culling the generation of unnecessary getters & setters, but it is gated on this change.

**Comment by yjbanov** [comment]
Since it's blocking you, let's bump it to P1 and find an owner tomorrow.

**Comment by kegluneq** [comment]
@yjbanov , this is now obsolete, right?

**Comment by yjbanov** [comment]
We [still use](https://github.com/angular/angular/blob/6b5d345a2b24a99b13882382d159302e64b6ed18/modules/angular2/src/core/linker/element_injector.ts#L206) getters/setters to support events and queries.

**Comment by kegluneq** [comment]
Ah, I wasn't sure what was meant by "in element injector". Yes, we're still using reflected getters & setters in those cases.

**Comment by mhevery** [comment]
This is obsolete, and no longer relevant / actionable. To keep our issues clean, we are aggressively closing them.

Latest code gen fixes this.
